src/App.tsx

```diff
@@ -1,61 +1,150 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { TodoList } from './components/TodoList';
+
+const getUserById = (id: number): User | null => {
+  const user = usersFromServer.find(({ id: userId }) => {
+    return userId === id;
+  });
+
+  return user || null;
+};
+
+const todos: Todo[] = todosFromServer.map(todo => {
+  const user = getUserById(todo.userId);
+
+  return {
+    ...todo,
+    user,
+  };
+});
 
 export const App = () => {
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [allTodos, setAllTodos] = useState(todos);
+  const [userError, setUserError] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+
+  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
```

Comment:
Consider renaming `handleTitleChange` to `onTitleChange` for better readability. 🤓